### DISCUSSION TITLE
Command-T 6.0.0 release series — general feedback

### DISCUSSION THREAD

**Discussion by wincent** [wincent/command-t#397] [Releases]
For those who prefer the discussion format, here's a place for feedback on any of the versions in the 6.0.x release series (this is a companion to #393 over in [issues](https://github.com/wincent/command-t/issues)). Copying over some of the same content from there:

# Command-T 6.0.0

- [6.0.0-a.0 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-a.0).
- [6.0.0-a.1 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-a.1).
- [6.0.0-a.2 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-a.2).
- [6.0.0-a.3 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-a.3).
- [6.0.0-a.4 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-a.4).
- [6.0.0-b.0 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-b.0).
- [6.0.0-b.1 Release notes](https://github.com/wincent/command-t/releases/tag/6.0.0-b.1).
- [`|command-t-upgrading|`](https://github.com/wincent/command-t/blob/6.0.0-b.1/doc/command-t.txt#L93).
- [Blog post, "Command-T v6.0 — the Lua rewrite"](https://wincent.com/blog/command-t-lua-rewrite).
- https://github.com/wincent/command-t/issues/391
- [`5-x-release` branch](https://github.com/wincent/command-t/tree/5-x-release) (if you wish to remain on the previous release serious).

Now, this is an _alpha_ release (see [`|command-t-known-issues|`](https://github.com/wincent/command-t/blob/6.0.0-b.1/doc/command-t.txt#L257)), a total rewrite, subject to change, incompletely documented, with known bugs and feature gaps. But, because the entire world installs Neovim plugins by pointing at `HEAD` of `main`, people are probably going to end up trying this whether they intend to or not.

So, this feedback thread is to provide a shared space for discussing the release.

**Comment by dfishburn** [comment]
I am running on Windows 10.
With the change to straight up POSIX, can I build on Windows?
```
command-t\command-t-6.0.0-a.4\lua\wincent\commandt\lib>make
process_begin: CreateProcess((null), getconf LONG_BIT, ...) failed.
make: *** windows32_openpipe (): unable to launch process (e=2)
.  Stop.
```
Using this compiler
```
"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\VsDevCmd.bat"
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.10.3
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
```

**Reply by wincent** [reply]
I haven't actually owned a Windows machine since v3.11, so other than the occasional exprience in a virtual machine, I've always been reliant on users to test and patch things for support on Windows. Given that I can't provide quality support for a platform that I don't have real access to, I decided to focus on POSIX in v6 of Command-T — this is not to say that I would reject a PR that added the necessary changes to support Windows, just that it's not within my means to be able to produce such a patch my self (and sadly, even to test it to the point of being able to really vouch for it).

So, I wouldn't rule out that this could be made to work in the future, but that the work to do so will really have to come from people with an investment in Windows. About then only condition I have is that I don't want support for multiple platforms to add an unacceptable level of complexity to the code base (where "unacceptable" is a subjective thing which I'm using to stand in for "something that would negatively impact my ability to maintain the code base at a reasonable cost").

The Ruby implementation is still in v6 on the `main` branch, though, and _might_ work with Windows (although with the same disclaimer: I've never been in a position to test Command-T properly on Windows, or even really provide proper documentation on it from first hand experience).
